### PR TITLE
Update guest framework

### PR DESCRIPTION
**Pull request by MarcelMue**
Prep for switching to KIND in https://github.com/giantswarm/aws-operator/pull/2003.

Adding the host k8s client helps us unpick the framework complexity. So we can eventually remove it completely.

### CODE REVIEW

**Comment by rossf7** [comment]
@MarcelMue I need to hijack this. OK if I assign myself?

File: pkg/framework/guest.go [hunk 1]
```diff
@@ -27,16 +24,19 @@ const (
 type GuestConfig struct {
 	Logger micrologger.Logger
 
+	HostK8sClient kubernetes.Interface
```
Comment (kopiczko) [inline pkg/framework/guest.go]: We don't have validation for this config in `NewGuest`

Comment (rossf7) [reply pkg/framework/guest.go]: Fixed.

**Comment by kopiczko** [comment]
How do you want to do the migration in other project using the package?

**Comment by rossf7** [comment]
> How do you want to do the migration in other project using the package?

There are still 7 other projects using the framework package. Several can be resolved by migrating them to KIND. I'm tracking that here and will merge once other projects are ready.

https://github.com/giantswarm/giantswarm/issues/4606#issuecomment-544389000

**Review by kopiczko — APPROVED** [review]

**Comment by rossf7** [comment]
PRs are prepared for aws-operator and azure-operator. I'll merge this once e2e passes.

**Comment by rossf7** [comment]
AWS e2e is green and Azure is mainly working and failures not related. So going with this.

**Pull request merged**